.github/actions/install-dashboards/action.yml

```diff
@@ -33,28 +34,41 @@ runs:
         ref: 'main'
         fetch-depth: 0
 
-    - run: mkdir -p plugins
-      working-directory: OpenSearch-Dashboards
-      shell: bash
-
     - uses: actions/checkout@v2
       with:
         path: ${{ steps.determine-plugin-directory.outputs.plugin-directory }}
 
-    - id: osd-version
+    - id: osd-version-linux
+      if: ${{ runner.os == 'Linux' }}
       continue-on-error: true
       run: |
-        echo "::set-output name=osd-version::$(cat package.json | jq '.opensearchDashboards.version' | cut -c 2-4)"
+        echo "::set-output name=osd-version-linux::$(cat package.json | jq '.opensearchDashboards.version' | cut -c 2-4)"
         echo "::set-output name=osd-x-version::$(cat package.json | jq '.opensearchDashboards.version' | cut -c 2-3)"
       working-directory: ${{ steps.determine-plugin-directory.outputs.plugin-directory }}
       shell: bash
 
-    - id: branch-switch-if-possible
+    - id: osd-version-windows
+      if: ${{ runner.os == 'Windows' }}
+      continue-on-error: true
+      run: |
+        echo "::set-output name=osd-version-linux::$(cat package.json | jq '.opensearchDashboards.version' | cut -c 2-4)"
+        echo "::set-output name=osd-x-version::$(cat package.json | jq '.opensearchDashboards.version' | cut -c 2-3)"
+      working-directory: ${{ steps.determine-plugin-directory.outputs.plugin-directory }}
+      shell: pwsh
+
+    - id: branch-switch-if-possible-linux
       continue-on-error: true # Defaults onto main if the branch switch doesn't work
-      if: ${{ steps.osd-version.outputs.osd-version }}
-      run: git checkout ${{ steps.osd-version.outputs.osd-version }} || git checkout ${{ steps.osd-version.outputs.osd-x-version }}x
+      if: ${{ runner.os == 'Linux' && steps.osd-version-linux.outputs.osd-version-linux }}
+      run: git checkout ${{ steps.osd-version-linux.outputs.osd-version }} || git checkout ${{ steps.osd-version-linux.outputs.osd-x-version }}x
       working-directory: ./OpenSearch-Dashboards
       shell: bash
+    
+    - id: branch-switch-if-possible-windows
+      continue-on-error: true # Defaults onto main if the branch switch doesn't work
+      if: ${{ runner.os == 'Windows' && steps.osd-version-linux.outputs.osd-version-windows }}
+      run: git checkout ${{ steps.osd-version-windows.outputs.osd-version }} || git checkout ${{ steps.osd-version-windows.outputs.osd-x-version }}x
+      working-directory: ./OpenSearch-Dashboards
+      shell: pwsh
```

Comment:
Are you seeing a difference between bash/pwsh when running these commands?  If it does, use `action-cond` [[link](https://stackoverflow.com/a/67291623/533057)] to set the platform specific shell for these commands.